pkg/util/testing/core.go

```diff
@@ -60,16 +60,23 @@ func CheckEventRecordedFor(ctx context.Context, k8sClient client.Client,
 		len(events.Items), eventReason, eventType, eventMessage, ref.Namespace)
 }
 
-// HasEventAppeared returns if an event has been emitted
-func HasEventAppeared(ctx context.Context, k8sClient client.Client, event corev1.Event) (bool, error) {
+// HasAnyEventAppeared returns if an event has been emitted
+func HasAnyEventAppeared(ctx context.Context, k8sClient client.Client, matcher func(*corev1.Event) bool) (bool, error) {
```

Comment:
I feel like this could be renamed to:
```suggestion
func HasMatchingEventAppeared(ctx context.Context, k8sClient client.Client, matcher func(*corev1.Event) bool) (bool, error) {
```

It immediately suggests that there is some matcher involved, while "any" can literally be read as "any event".

Reply:
Done.